### PR TITLE
[WIP] Abstract component-wise matching

### PR DESCRIPTION
**Pull request by ivansg44**
This addresses #9.

Similar code is used in 1-5 gram component matching. This code should be abstracted into a single function.

### CODE REVIEW

File: lexmapr/pipeline.py [hunk 1]
```diff
@@ -966,15 +966,15 @@ def get_gram_chunks(num):
...
-                                    localTrigger = True
+                                    if (sampleRevisedWithSuffix in resource_terms_revised.keys() and not localTrigger):  # Not trigger true is used here -reason
+                                        # resourceId = resourceRevisedTermsDict[sampleRevisedWithSuffix]
+                                        partialMatchedList.append(sampleRevisedWithSuffix)
+                                        status_addendum.append("Suffix Addition- " + suffixString + " to the Input")
+                                        for eachTkn in grmTokens:
+                                            covered_tokens.append(eachTkn)
+                                            if eachTkn in remaining_tokens:
+                                                remaining_tokens.remove(eachTkn)
+                                        localTrigger = True
```
Comment (ivansg44) [inline lexmapr/pipeline.py]: @lexmapr Can you confirm this was an indentation typo? This if-statement is duplicated 5 times for 1-5 component matching, and it is only indented to the same level of the for-loop in the 2-component matching section.

Changing this indentation effected one test: test_full_term_sfx_match.

File: lexmapr/pipeline.py [hunk 2]
```diff
@@ -966,15 +966,15 @@ def get_gram_chunks(num):
...
-                                if (sampleRevisedWithSuffix in resource_terms_revised.keys() and not localTrigger):  # Not trigger true is used here -reason
-                                    # resourceId = resourceRevisedTermsDict[sampleRevisedWithSuffix]
-                                    partialMatchedList.append(sampleRevisedWithSuffix)
-                                    status_addendum.append("Suffix Addition- " + suffixString + " to the Input")
-                                    for eachTkn in grmTokens:
-                                        covered_tokens.append(eachTkn)
-                                        if eachTkn in remaining_tokens:
-                                            remaining_tokens.remove(eachTkn)
-                                    localTrigger = True
+                                    if (sampleRevisedWithSuffix in resource_terms_revised.keys() and not localTrigger):  # Not trigger true is used here -reason
```
Comment (ivansg44) [inline lexmapr/pipeline.py]: @lexmapr Can you confirm this was an indentation typo? This if-statement is duplicated 5 times for 1-5 component matching, and it is only indented to the same level of the for-loop in the 2-component matching section.

Changing this indentation effected one test: test_full_term_sfx_match.

Comment (dfornika) [reply lexmapr/pipeline.py]: I think it is a typo and I may have introduced it while creating the `pipeline.run()` method. You can see the original structure of this section of code here:

https://github.com/lexmapr/LexMapr/commit/de1238df9da862d0f2d3584b827dbb7e7f3ee58c#diff-43bdea91548ee972633b982f54d60247R1158

**Comment by ivansg44** [comment]
@dfornika Can you review this PR?

**Comment by ivansg44** [comment]
Ready for a review. I pulled find_component_match out of run. Similar to find_full_term_match, It still has two parameters that should be eliminated (cleaned_sample and status_addendum), but again this can be done in the future after the implementation of a function for preprocessing. @dfornika

**Pull request merged**